controllers/machine_controller_test.go

```diff
@@ -1397,6 +1465,28 @@ func TestIsDeleteNodeAllowed(t *testing.T) {
 			emp.SetName("test-cluster")
 			emp.SetNamespace("test-cluster")
 
+			mcpBeingDeleted := &unstructured.Unstructured{
+				Object: map[string]interface{}{},
+			}
+			mcpBeingDeleted.SetAPIVersion("controlplane.cluster.x-k8s.io/v1alpha4")
+			mcpBeingDeleted.SetKind("AWSManagedControlPlane")
+			mcpBeingDeleted.SetName("test-cluster-2")
+			mcpBeingDeleted.SetNamespace("test-cluster")
+			mcpBeingDeleted.SetDeletionTimestamp(&metav1.Time{Time: time.Now()})
+
+			empBeingDeleted := &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"status": map[string]interface{}{
+						"externalManagedControlPlane": true,
+					},
+				},
+			}
+			empBeingDeleted.SetAPIVersion("controlplane.cluster.x-k8s.io/v1alpha4")
+			empBeingDeleted.SetKind("AWSManagedControlPlane")
+			empBeingDeleted.SetName("test-cluster-3")
+			empBeingDeleted.SetNamespace("test-cluster")
+			empBeingDeleted.SetDeletionTimestamp(&metav1.Time{Time: time.Now()})
```

Comment:
nit
Those vars could be created out of the for loop

Reply:
updated

Reply:
thanks!